examples/profile.js

```diff
@@ -123,4 +210,24 @@ app.MainController = function($http, $scope) {
 };
 
 
+/**
+ * @param {ol.Coordinate} coordinate The current pointer coordinate.
+ * @param {ol.geom.Geometry|undefined} geometry The geometry to snap to.
+ */
+app.MainController.prototype.snapToGeometry = function(coordinate, geometry) {
+  var closestPoint = geometry.getClosestPoint(coordinate);
+  // compute distance to line in pixels
+  var dx = closestPoint[0] - coordinate[0];
+  var dy = closestPoint[1] - coordinate[1];
+  var dist = Math.sqrt(dx * dx + dy * dy);
+  var pixelDist = dist / this['map'].getView().getResolution();
+
+  if (pixelDist < 8) {
+    this['profile'].highlight(closestPoint[2]);
+  } else {
+    this['profile'].highlight(-1);
+  }
+};
```

Comment:
I see there is still an issue when `highlight()` is called while we are in a digest cycle: the wrapped hoverCallback will call `$apply()` again. I think it would be better to remove the wrapper and instead have the profile know about `$apply()`. Then it would be able to call apply only when handling a browser events and in the other cases, it would be assumed that the user is in a digest cycle or will apply herself.
We can use a workaround to hide the scope from the profile function, but why would we want to hide it?